### ISSUE TITLE
`Calicoctl endpoint profile (append...)` has unexpected output when no profile is specified

### ISSUE THREAD

**Issue by TrimBiggs** [projectcalico/calico-docker#557]
The `calicoctl endpoint <endpoint_id> profile (append|set|remove)` command is written in docopt as:
```
calicoctl endpoint <ENDPOINT_ID> profile (append|remove|set) [--host=<HOSTNAME>] 
    [--orchestrator=<ORCHESTRATOR_ID>] [--workload=<WORKLOAD_ID>]  [<PROFILES>...]
```

This allows the user to ignore the `<PROFILES>` argument entirely.  This has strange output:
```
# This does not actually add any profiles.
$ calicoctl endpoint d79123c4784511e5bd1a080027f532f6 profile append
Profiles  appended

$ calicoctl endpoint d79123c4784511e5bd1a080027f532f6 profile show
+------+
| Name |
+------+
| PROF |
+------+

# This doesn't remove any profiles, but still succeeds.
$ calicoctl endpoint d79123c4784511e5bd1a080027f532f6 profile remove
Profiles  removed.

$ calicoctl endpoint d79123c4784511e5bd1a080027f532f6 profile show
+------+
| Name |
+------+
| PROF |
+------+

# This sets the profile to be an empty string (no profiles)
$ calicoctl endpoint d79123c4784511e5bd1a080027f532f6 profile set
Profiles  set.

$ calicoctl endpoint d79123c4784511e5bd1a080027f532f6 profile show
Endpoint has no profiles associated with it.
```

The `<PROFILES>` param should be required.

**Comment by TrimBiggs** [comment]
Actually, it doesn't really make much sense to have the other flags in this command either...  

I imagine this command used to be merged with the `endpoint <ENDPOINT_ID> profile show command`, which would explain this and why the `<PROFILES>` param is optional here.

Something like this:
```
calicoctl endpoint <ENDPOINT_ID> profile (append|remove|set|show) [--host=<HOSTNAME>] 
    [--orchestrator=<ORCHESTRATOR_ID>] [--workload=<WORKLOAD_ID>]  [<PROFILES>...]
```

**Comment by TrimBiggs** [comment]
The reason the `<PROFILES>` param is optional is to allow the `set` command to use an empty list of profiles to remove all profiles.

The change should be to update the output of the commands to be more clear what is going on.

**Comment by TrimBiggs** [comment]
Fixed by #568
